In [4]:
buffett <- read.csv("Factor_Data_SA_6.xls")

In [5]:
head(buffett)

Date       Mom    Mkt_rf  SMB     HML    RF     QMJ     BAB    Brk_ret
1 11/30/1976 0.0290  0.0036  0.0231 0.0152 0.0040 -0.0201 0.0045  0.1544
2 12/31/1976 0.0078  0.0565  0.0300 0.0223 0.0040 -0.0089 0.0317  0.1465
3 1/31/1977  0.0397 -0.0405  0.0478 0.0427 0.0036 -0.0263 0.0407  0.0000
4 2/28/1977  0.0038 -0.0195  0.0108 0.0051 0.0035  0.0207 0.0249  0.0000
5 3/31/1977  0.0049 -0.0137  0.0099 0.0100 0.0038 -0.0085 0.0123  0.0778
6 4/30/1977  0.0424  0.0015 -0.0012 0.0344 0.0038 -0.0173 0.0106 -0.0103
  Brk_exret Mkt    
1  0.1504    0.0076
2  0.1425    0.0605
3 -0.0036   -0.0369
4 -0.0035   -0.0160
5  0.0740   -0.0099
6 -0.0141    0.0053

In [6]:
summary(buffett)

     Date                Mom                Mkt_rf               SMB           
 Length:500         Min.   :-0.343900   Min.   :-0.232400   Min.   :-0.172800  
 Class :character   1st Qu.:-0.009475   1st Qu.:-0.019200   1st Qu.:-0.014225  
 Mode  :character   Median : 0.007800   Median : 0.010600   Median : 0.002100  
                    Mean   : 0.006506   Mean   : 0.006447   Mean   : 0.002327  
                    3rd Qu.: 0.028725   3rd Qu.: 0.034925   3rd Qu.: 0.020150  
                    Max.   : 0.183300   Max.   : 0.124700   Max.   : 0.221400  
      HML                  RF                QMJ                 BAB           
 Min.   :-0.111000   Min.   :0.000000   Min.   :-0.090700   Min.   :-0.154000  
 1st Qu.:-0.013075   1st Qu.:0.000975   1st Qu.:-0.008125   1st Qu.:-0.004375  
 Median : 0.000800   Median :0.003900   Median : 0.004100   Median : 0.011050  
 Mean   : 0.002637   Mean   :0.003711   Mean   : 0.004397   Mean   : 0.008972  
 3rd Qu.: 0.017125   3rd Qu.:0.005325   

In [8]:
# Install required packages if not already installed
# install.packages("tidyverse")
# install.packages("ggplot2")

# Load required libraries
library(tidyverse)
library(ggplot2)

In [11]:
# Calculate excess returns
buffett$ExcessRet <- buffett$Brk_ret - buffett$RF
buffett$ExcessMkt <- buffett$Mkt - buffett$RF

In [12]:
# Perform the regression analysis
model <- lm(ExcessRet ~ ExcessMkt, data = buffett)

# Print the regression results
summary(model)


Call:
lm(formula = ExcessRet ~ ExcessMkt, data = buffett)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.17263 -0.03475 -0.00688  0.02608  0.33062 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 0.010829   0.002724   3.976 8.05e-05 ***
ExcessMkt   0.689755   0.061777  11.165  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.06025 on 498 degrees of freedom
Multiple R-squared:  0.2002,	Adjusted R-squared:  0.1986 
F-statistic: 124.7 on 1 and 498 DF,  p-value: < 2.2e-16


In [13]:
alpha <- coef(summary(model))[1, 1]

# Print Jensen's Alpha
print(alpha)

[1] 0.01082887


In [14]:
# Perform the regression analysis
model2 <- lm(ExcessRet ~ ExcessMkt + SMB + HML, data = buffett)

# Print the regression results
summary(model2)


Call:
lm(formula = ExcessRet ~ ExcessMkt + SMB + HML, data = buffett)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.14769 -0.03547 -0.01046  0.02657  0.32709 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  0.009479   0.002656   3.569 0.000394 ***
ExcessMkt    0.819781   0.062958  13.021  < 2e-16 ***
SMB         -0.292096   0.090797  -3.217 0.001380 ** 
HML          0.451676   0.096140   4.698 3.41e-06 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.05806 on 496 degrees of freedom
Multiple R-squared:  0.2603,	Adjusted R-squared:  0.2558 
F-statistic: 58.18 on 3 and 496 DF,  p-value: < 2.2e-16


The coefficient of HML in the three-factor model regression for Berkshire Hathaway's excess returns represents the estimated effect of the HML (Value Factor) on Berkshire Hathaway's excess returns, while controlling for the other factors in the model (MKT-rf and SMB).

In this case, if the coefficient of HML is 0.451676, it means that for every one-unit increase in HML, Berkshire Hathaway's excess returns are estimated to increase by 0.451676 units, assuming the other factors remain constant.

Furthermore, the sign of the coefficient can provide additional insight. Since the coefficient is positive (assuming it's statistically significant), it suggests that when the HML factor (representing the value premium) is high, Berkshire Hathaway's excess returns tend to be higher than expected, considering the influence of MKT-rf and SMB.

However, it's important to interpret the coefficient in the context of the specific dataset and the model's assumptions. Additionally, be mindful of potential multicollinearity and the statistical significance of the coefficient, which can be assessed through the p-value associated with the coefficient in the regression output.

The coefficient of 0.451676 for HML in the three-factor model regression suggests that Berkshire Hathaway's excess returns have a positive association with the HML (Value Factor).

In the Fama-French three-factor model, HML represents the excess return of value stocks relative to growth stocks. A positive coefficient indicates that Berkshire Hathaway's excess returns tend to be higher when the value factor is high, implying a tilt towards value stocks.

Therefore, based on the coefficient value, it suggests that Berkshire Hathaway's investment strategy has a preference for value stocks rather than growth stocks. This aligns with the known investment philosophy of Warren Buffett, who is known for his value investing approach, focusing on undervalued companies with strong fundamentals.

However, it's important to note that this interpretation relies on the specific model and dataset used, and it's crucial to consider other factors and perform a thorough analysis to confirm the investment style or strategy of Berkshire Hathaway.

---

The coefficient of -0.292096 for SMB in the three-factor model regression suggests that Berkshire Hathaway's excess returns have a negative association with the SMB (Size Factor).

In the Fama-French three-factor model, SMB represents the excess return of small-cap stocks relative to large-cap stocks. A negative coefficient indicates that Berkshire Hathaway's excess returns tend to be higher when the size factor is low, implying a tilt away from small-cap stocks.

Therefore, based on the coefficient value, it suggests that Berkshire Hathaway's investment strategy is tilted towards large-cap stocks rather than small-cap stocks. This aligns with Berkshire Hathaway's historical focus on acquiring established, well-known companies with substantial market capitalization.

It's important to note that this interpretation is based on the specific model and dataset used, and further analysis and consideration of other factors are needed to fully understand Berkshire Hathaway's investment strategy regarding small-cap and large-cap stocks.

Warren Buffett’s level of outperformance based on the three factor model is:

0.9% per month

---

In [16]:

# Perform the regression analysis
model3 <- lm(ExcessRet ~ ExcessMkt + SMB + HML + Mom + BAB + QMJ, data = buffett)

# Print the regression results
summary(model3)



Call:
lm(formula = ExcessRet ~ ExcessMkt + SMB + HML + Mom + BAB + 
    QMJ, data = buffett)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.14561 -0.03317 -0.00769  0.02345  0.32308 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  0.004654   0.002843   1.637 0.102200    
ExcessMkt    0.936662   0.073590  12.728  < 2e-16 ***
SMB         -0.175346   0.101652  -1.725 0.085160 .  
HML          0.399840   0.109863   3.639 0.000302 ***
Mom         -0.048167   0.065350  -0.737 0.461440    
BAB          0.274218   0.086524   3.169 0.001623 ** 
QMJ          0.406937   0.155086   2.624 0.008961 ** 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.05694 on 493 degrees of freedom
Multiple R-squared:  0.2928,	Adjusted R-squared:  0.2842 
F-statistic: 34.02 on 6 and 493 DF,  p-value: < 2.2e-16
